common/utils/include/claragenomics/utils/buffer.hpp

```diff
@@ -185,15 +200,16 @@ class buffer
         resize(new_size, _stream);
     }
 
-    /// @brief This method swaps the contents of this buffer with another buffer.
-    /// @param v The buffer with which to swap.
-    void swap(buffer& v)
+    /// @brief This method swaps the contents of two buffers.
+    /// @param a One buffer.
+    /// @param b The other buffer.
+    friend void swap(buffer& a, buffer& b)
     {
-        std::swap(_data, v._data);
```

Comment:
Add `noexcept` to swap.
While not critical here, I would not enforce `std::swap` in the implementation, but do
```
using std::swap;
swap(_size, v._size);
...
```
See https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ro-custom. They actually use exactly this example. :D

Reply:
Cool, didn't know that!

done

Reply:
:)